cmd/api/src/model/auth.go

```diff
@@ -460,18 +460,13 @@ type User struct {
 	Unique
 }
 
-func (s User) AuditData() AuditData {
-	data := AuditData{
-		"id":             s.ID,
-		"principal_name": s.PrincipalName,
-		"roles":          s.Roles.IDs(),
-	}
-
-	if s.SAMLProviderID.Valid {
-		data["saml_provider_id"] = s.SAMLProviderID
+func (s *User) AuditData() AuditData {
+	return AuditData{
+		"id":               s.ID,
+		"principal_name":   s.PrincipalName,
+		"roles":            s.Roles.IDs(),
+		"saml_provider_id": s.SAMLProviderID.ValueOrZero(),
```

Comment:
This ensures it will show up in the `fields` section of an `audit_log` entry. It will help us determine if the `UpdateUser` call changed the field, which is commonly performed by some of the SAML Provider endpoints.